src/ophyd_async/core/_signal.py

```diff
@@ -56,7 +56,15 @@ def __init__(self, backend: SignalBackend):
         self.backend = self._init_backend = backend
 
     async def connect_mock(self, device: Device, mock: LazyMock):
-        self.backend = MockSignalBackend(self._init_backend, mock)
+        # TODO: ugly circular import
+        from ._derived_signal import DerivedSignalBackend
+
+        if isinstance(self._init_backend, DerivedSignalBackend):
+            # It makes no sense to mock a derived signal backend, so
+            # use the real thing which will be backed by mocked signals
+            self.backend = self._init_backend
```

Comment:
Doesn't it? We still want mock utilities...

Is there a way to do it [here](https://github.com/bluesky/ophyd-async/blob/bdffa0111d4e7c432ce0ce8f3214b09e81f526ac/src/ophyd_async/core/_mock_signal_backend.py#L27-L34) like we do with soft signal backend?

Reply:
The problem is what does it mean to call `set_mock_value` on a `DerivedSignal`? That makes it more of a "half mocked" signal, which might be confusing...

Reply:
Yea, I was thinking of maybe allowing it and putting an assert here

https://github.com/bluesky/ophyd-async/blob/93ac49901f6b32e5a8243a278119c9ced53dc7b4/src/ophyd_async/testing/_mock_signal_utils.py#L35-L38

but we probably don't have a use case really.